plugins/device/rest.go

```diff
@@ -58,9 +60,9 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if deviceutils.DeviceInList(req.Device, devices) {
+	if index := deviceutils.DeviceInList(req.Device, devices); index == -1 {
```

Comment:
Shouldn't this be `!= -1` ?

Also, in general, use the `<` operator and not a specific value. A specific value indicates that there might be more such values with specific meanings